packages/web-api-scan-runner/src/crawl-runner/crawl-runner.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { GlobalLogger } from 'logger';
+import { inject, injectable } from 'inversify';
+import { Crawler, iocTypes as crawlerIocTypes, SimpleCrawlerRunOptions } from 'accessibility-insights-crawler';
+import { Page } from 'puppeteer';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+
+type CrawlerProvider = () => Promise<Crawler<string[]>>;
+
+@injectable()
+export class CrawlRunner {
+    constructor(
+        @inject(crawlerIocTypes.CrawlerProvider) private readonly getCrawler: CrawlerProvider,
+        @inject(GlobalLogger) private readonly logger: GlobalLogger,
+        @inject(ScanMetadataConfig) scanMetadataConfig: ScanMetadataConfig,
+    ) {
+        const scanMetadata = scanMetadataConfig.getConfig();
+        this.logger.setCommonProperties({ scanId: scanMetadata.id });
+    }
+
+    public async run(baseUrl: string, discoveryPatterns: string[], page: Page): Promise<string[] | undefined> {
+        const crawler = await this.getCrawler();
+        if (crawler == null) {
+            this.logger.logInfo('No crawler provided by crawler provider');
+
+            return undefined;
+        }
+
+        const crawlerRunOptions = {
+            baseUrl,
+            discoveryPatterns,
+            page,
```

Comment:
```suggestion
            page.getUnderlyingPage(),
```
Uses the puppeteer page instead of our wrapper